routers/api/v1/repo/file.go

```diff
@@ -45,3 +45,27 @@ func GetArchive(ctx *context.APIContext) {
 
 	repo.Download(ctx.Context)
 }
+
+func GetEditorconfig(ctx *context.APIContext) {
+	if !ctx.Repo.HasAccess() {
+		ctx.Status(404)
+		return
+	}
+	ec, err := ctx.Repo.GetEditorconfig()
+	if err != nil {
+		if git.IsErrNotExist(err) {
+			ctx.Handle(404, "ErrEditorconfigNotExists", err)
+		} else {
+			ctx.Handle(500, "ErrGettingEditorconfig", err)
+		}
+		return
+	}
+
+	fileName := ctx.Params("*")
```

Comment:
Why it is `fileName`? The UI actually passes a branch name? (`{{$.Repository.DefaultBranch}}`)